spring-integration-core/src/test/java/org/springframework/integration/dsl/routers/RouterTests.java

```diff
@@ -899,6 +911,32 @@ public IntegrationFlow propagateErrorFromGatherer(TaskExecutor taskExecutor) {
 					.get();
 		}
 
+		@Bean
+		public IntegrationFlow routerWithSubflowStartsWithAChannel() {
+			return IntegrationFlows.from(MessageChannels.direct())
+					.route("'foo'", mapping -> mapping
+							.subFlowMapping("foo",  subflowStartsWithAChannel()))
```

Comment:
This doesn't reflect the use-case in the SO question.
The real point is that it is used as a *lambda*, so we don't have a `StandardIntegrationFlow` instance there yet...

Reply:
This is still a lambda (`IntegrationFlow`).

Reply:
We have a test config like this:
```
@Bean
		public IntegrationFlow upperCase() {
			return f -> f
					.<String>handle((p, h) -> p.toUpperCase());
		}

		@Bean
		public IntegrationFlow routeSubflowToReplyChannelFlow() {
			return f -> f
					.<Boolean>route("true", m -> m
							.subFlowMapping(true, upperCase())
					);
		}

```

And over some AOP magic I ended up in the `StandardIntegrationFlow.getInputChannel()` which returns for me an `upperCase.input` `DirectChannel`.
That's all just because we call a *bean method* which is going to resolve to a bean and we have there `IntegrationFlowLifecycleAdvice` wrapping a lambda and delegate to the created internally `StandardIntegrationFlow`.

I understand that we will still have there a `bridge` if you declare a `channel()` in the beginning of that lambda, but that's already all by design.
If you would like to avoid that, please, use `IntegrationFlows.from([CHANNEL_NAME])`.

So, I still insist that we need to end up with the documentation improvement only.